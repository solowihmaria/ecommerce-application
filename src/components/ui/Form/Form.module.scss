@use '../../../assets/styles/constants.module' as c;

.form {
    display: flex;
    flex-direction: column;
    gap: 1.5rem;
    align-items: center;
    justify-content: center;
    background-color: c.$color-primary-dark;
}
