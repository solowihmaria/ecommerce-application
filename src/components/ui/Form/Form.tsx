import React from 'react';
import type { FormProps } from './Form.types';
import styles from './Form.module.scss';

export const Form = ({
    children,
    className = '',
    ...props
}: FormProps): React.JSX.Element => {
    return (
        <form className={`${styles.form} ${className}`} {...props}>
            {children}
        </form>
    );
};
