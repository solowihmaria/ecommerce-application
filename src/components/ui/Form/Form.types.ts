import type { FormHTMLAttributes, ReactNode } from 'react';

export interface FormProps extends FormHTMLAttributes<HTMLFormElement> {
    children: ReactNode;
    className?: string;
}
