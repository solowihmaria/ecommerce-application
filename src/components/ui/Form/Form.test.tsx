import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Form } from './Form';

describe('<Form/> component', () => {
    test('Render Form component without any properties', () => {
        const TestComponent = () => {
            return (
                <>
                    <input placeholder="input1"></input>
                    <input placeholder="input2"></input>
                </>
            );
        };

        render(
            <Form>
                <TestComponent />
            </Form>
        );
        const form = screen.getByTestId('form-test-id');
        expect(form).toBeInTheDocument();
        expect(form).toMatchSnapshot();
    });
});
