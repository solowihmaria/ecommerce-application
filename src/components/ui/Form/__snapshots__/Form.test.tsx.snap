// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form/> component Render Form component without any properties 1`] = `
<form
  class="form"
  data-testid="form-test-id"
>
  <input
    placeholder="input1"
  />
  <input
    placeholder="input2"
  />
</form>
`;
